x/group/server/query_server.go

```diff
@@ -9,6 +9,24 @@ import (
 	"github.com/regen-network/regen-ledger/x/group"
 )
 
+func (s serverImpl) Groups(ctx types.Context, request *group.QueryGroupsRequest) (*group.QueryGroupsResponse, error) {
+	it, err := s.groupTable.PrefixScan(ctx, nil, nil)
```

Comment:
You have to use `request.Pagination.Key` in `PrefixScan` `start` parameter. Check out `Paginate` function documentation for more details on how to instantiate iterators to pass into `Paginate`.